prime-router/docs/design/hl7-validation/design.md

```diff
@@ -0,0 +1,115 @@
+# Problem Statement
+As a sender/receiver I want submitted/received data to be validated to ensure data integrity and quality.
+
+There needs to exist a collection of settings and reusable components to perform varying types of validation at various points in the HL7 message's lifecycle.
```

Comment:
You mention "various types of validation" and "various points". Perhaps you could expound here on what those exactly are?